### ISSUE TITLE
Typescript parser has a deferred require

### ISSUE THREAD

**Issue by Downchuck** [prettier/prettier#11269]
Over in `src/language-js/parse/typescript.js`:

The `parse` function has this deferred require  (Line 40) which would be better placed at the top of the script:
```
  const { parseWithNodeMaps } = require("@typescript-eslint/typescript-estree");
```

**Comment by fisker** [comment]
It is intentionally, all core parsers require in `parse`.

**Comment by Downchuck** [comment]
Apologies for the poorly researched report. There's still something off in the require chain:
```
import prettier from "prettier/standalone";
import parserBabel from "prettier/parser-babel";
```

That's hitting:
```
./node_modules/prettier/parser-typescript.js
Critical dependency: the request of a dependency is an expression
```

Seems like there was a fix for issue #4959 as far as style went, for someone in a different module, and it was related to the require style in use.  Still I am not clear if that parser should have been triggered anyway.  If I figure it out I'll post a PR. It's  unusual behavior - given it's the only warning I'm seeing and is stylistically odd to trigger.

**Comment by fisker** [comment]
This should be fixed by https://github.com/prettier/prettier/pull/11130, haven't released yet.
